题目是：某种感冒冲剂规定每包质量为12g，质量不足或质量过大都会引发严重后果。分析过去的生产数据可知，标准差为2g，质检员抽取25包冲剂称重检验，平均每包的质量为11.85g。假定感冒冲剂质量服从正态分布。取显著性水平为α=0.05，检验每包感冒冲剂的质量是否符合标准要求。


In [ ]:
import numpy as np
from scipy import stats

# 已知条件
mu0 = 12      # 规定标准质量（总体均值）
sigma = 2     # 总体标准差（已知）
n = 25        # 样本量
x_bar = 11.85 # 样本均值
alpha = 0.05  # 显著性水平

# 计算检验统计量 Z
Z = (x_bar - mu0) / (sigma / np.sqrt(n))
print("检验统计量 Z =", round(Z, 4))

# 计算临界值（双侧检验）
z_critical = stats.norm.ppf(1 - alpha/2)
print("临界值 Z_alpha/2 = ±", round(z_critical, 4))

# 计算 p 值（双侧检验）
p_value = 2 * (1 - stats.norm.cdf(np.abs(Z)))
print("p 值 =", round(p_value, 6))

# 做出决策
print("\n=== 检验结果 ===")
if np.abs(Z) > z_critical:
    print("拒绝原假设 H0（|Z| =", round(np.abs(Z), 4), "> Z临界值 =", round(z_critical, 4), ")")
    print("结论：在 alpha = 0.05 的显著性水平下，认为每包感冒冲剂的质量不符合标准要求。")
else:
    print("不拒绝原假设 H0（|Z| =", round(np.abs(Z), 4), "<= Z临界值 =", round(z_critical, 4), ")")
    print("结论：在 alpha = 0.05 的显著性水平下，没有足够证据表明每包感冒冲剂的质量不符合标准要求。")

if p_value < alpha:
    print("\np 值判断：拒绝原假设（p =", round(p_value, 6), "< alpha =", alpha, ")")
else:
    print("\np 值判断：不拒绝原假设（p =", round(p_value, 6), ">= alpha =", alpha, ")")

检验统计量 Z = -0.375
临界值 Z_alpha/2 = ± 1.96
p 值 = 0.70766

=== 检验结果 ===
不拒绝原假设 H0（|Z| = 0.375 <= Z临界值 = 1.96 )
结论：在 alpha = 0.05 的显著性水平下，没有足够证据表明每包感冒冲剂的质量不符合标准要求。

p 值判断：不拒绝原假设（p = 0.70766 >= alpha = 0.05 )


要解决这道假设检验问题，我们需要通过**假设检验**的步骤来判断感冒冲剂的质量是否符合标准要求。以下是详细的数学原理和步骤：
### **1. 明确问题与假设**
- **目标**：检验每包感冒冲剂的质量是否符合标准（即总体均值是否为12g）。
- **已知条件**：
  - 总体标准差 \( \sigma = 2 \, \text{g} \)（已知，因生产数据可得）；
  - 样本量 \( n = 25 \)；
  - 样本均值 \( \bar{x} = 11.85 \, \text{g} \)；
  - 显著性水平 \( \alpha = 0.05 \)；
  - 总体服从正态分布（题目假设）。
- **原假设与备择假设**：
  - 原假设 \( H_0: \mu = 12 \, \text{g} \)（质量符合标准）；
  - 备择假设 \( H_1: \mu \neq 12 \, \text{g} \)（质量不符合标准，双侧检验）。
### **2. 选择检验方法：Z检验**
由于**总体标准差已知**且**总体服从正态分布**，即使样本量 \( n = 25 \)（小于30），仍可使用**Z检验**（若总体标准差未知，需用t检验，但本题中 \( \sigma \) 已知，故用Z检验）。  
检验统计量公式为：  
\[
Z = \frac{\bar{x} - \mu_0}{\sigma / \sqrt{n}}
\]  
其中，\( \mu_0 = 12 \, \text{g} \) 为原假设的总体均值，\( \sigma = 2 \, \text{g} \) 为总体标准差，\( n = 25 \) 为样本量。
### **3. 计算检验统计量**
代入已知数值：  
\[
Z = \frac{11.85 - 12}{2 / \sqrt{25}} = \frac{-0.15}{2 / 5} = \frac{-0.15}{0.4} = -0.375
\]
### **4. 确定临界值（双侧检验）**
显著性水平 \( \alpha = 0.05 \)，双侧检验的临界值为 \( Z_{\alpha/2} \) 和 \( -Z_{\alpha/2} \)。  
查标准正态分布表（或使用统计软件），得 \( Z_{0.025} = 1.96 \)，因此临界值为 \( \pm 1.96 \)。
### **5. 决策规则**
- 若 \( |Z| > Z_{\alpha/2} \)，拒绝 \( H_0 \)（认为质量不符合标准）；
- 若 \( |Z| \leq Z_{\alpha/2} \)，不拒绝 \( H_0 \)（认为质量符合标准）。
### **6. 结论**
计算得 \( |Z| = 0.375 \)，小于临界值 \( 1.96 \)，因此**不拒绝原假设 \( H_0 \)**，即认为每包感冒冲剂的质量符合标准要求。
### **数学原理总结**
- **假设检验的核心**：通过样本数据推断总体参数是否与假设值一致，基于“小概率事件原理”（若原假设成立，检验统计量落入拒绝域的概率为 \( \alpha \)，视为小概率事件，拒绝原假设）。
- **Z检验的适用条件**：总体正态分布且总体标准差已知（或大样本下中心极限定理适用）。
- **双侧检验**：关注总体均值是否“不等于”假设值，临界值为正负对称值，拒绝域在分布两端。
**答案：** 每包感冒冲剂的质量符合标准要求（不拒绝 \( H_0 \)）。
